lib/codegen/src/isa/encoding.rs

```diff
@@ -78,12 +80,23 @@ impl fmt::Display for DisplayEncoding {
     }
 }
 
+type SizeCalculatorFn = fn(&RecipeSizing, Inst, &RegDiversions, &Function) -> u8;
+
+/// Returns the base size of the Recipe, assuming it's fixed. This is the default for most
+/// encodings.
```

Comment:
It'd be good to explicitly mention that encodings may be longer than this, depending on what registers they use.